avocado/utils/ssh.py

```diff
@@ -29,6 +29,45 @@
     SSH_CLIENT_BINARY = None
 
 
+class SessionError(Exception):
+    """
+    Represents an error with the SSH session.
+    """
+    def __init__(self, session):
+        """
+        :param session: the session object which occurred the error.
+        :type session: :class:`Session`
+        """
+        super(SessionError, self).__init__('SSH session error')
+        self.session = session
+
```

Comment:
A general session error might be useful I hope we can keep it to one or once again we will go in direction of reimplementing things.